cylc/flow/workflow_files.py

```diff
@@ -1337,6 +1351,44 @@ def validate_source_dir(source, flow_name):
     check_flow_file(source, logger=None)
 
 
+def get_sym_dirs(symlink_dirs: str) -> Dict[str, Dict[str, Any]]:
+    """Converts command line entered symlink dirs to a dictionary.
+
+    Args:
+        symlink_dirs (str): As entered by user on cli,
+                            e.g. "log=$DIR, share=$DIR2".
+        fill_syms (bool): If True, will fill missing
+
+    Raises:
+        WorkflowFilesError: If directory to be symlinked is not in permitted
+                            dirs: run, log, share, work, share/cycle
+
+    Returns:
+        dict: In the same form as would be returned by global config.
+            e.g. {'localhost': {'log': '$DIR',
+                                'share': '$DIR2'
+                                }
+                }
+    """
+    # Ensures the same nested dict format which is returned by the glb cfg
+    symdict: Dict[str, Dict[str, Any]] = {'localhost': {'run': None}}
+    if symlink_dirs == "":
+        return symdict
+    symlist = symlink_dirs.replace(" ", "").strip(',').split(',')
+    for pair in symlist:
+        try:
+            key, val = pair.split("=")
+        except ValueError:
+            continue
```

Comment:
Better to fail this as a command line error (it's just a command, so can be redone easily with no typos in it)?